src/plugins/core/sheet.ts

```diff
@@ -651,6 +651,9 @@ export class SheetPlugin extends CorePlugin<SheetState> implements SheetState {
     const { orderedSheetIds, sheets } = this;
     const name = cmd.name && cmd.name.trim().toLowerCase();
 
+    if (name && name.length > 31) {
```

Comment:
You can create a constant `MAX_SHEET_NAME_LENGTH = 31` and use it here and interpolated in the error message.

Small question: does 31 come from somewhere or did you chose it arbitrarily (Gsheet seems to have chosen 51)